src/store/Dashboard.js

```diff
@@ -176,23 +181,36 @@ const mutations = {
     state.assetList = currencies
   },
 
-  [types.GET_PRICE_BY_FILTER_REQUEST] (state) {},
+  [types.GET_PRICE_BY_FILTER_REQUEST] (state) {
+    Vue.set(state.assetChart, 'isLoading', true)
+  },
 
   [types.GET_PRICE_BY_FILTER_SUCCESS] (state, data) {
     Vue.set(state.assetChart, 'data', data)
+    Vue.set(state.assetChart, 'isLoading', false)
   },
 
   [types.GET_PRICE_BY_FILTER_FAILURE] (state, err) {
+    if (state.assetChart) {
+      state.assetChart.isLoading = false
```

Comment:
Replace with ```Vue.set()```. I guess this `if` statement is unnecessary.

Reply:
testErrorHandling doesn't work without if. Or maybe we need to change testErrorHandling function?

Reply:
Hm, rewrite this function seems to be good idea. Maybe provide `state` as second argument? Also don't access to types directly, please use object `types` that contains them.

`'GET_PORTFOLIO_HISTORY_FAILURE'` -> `types.GET_PORTFOLIO_HISTORY_FAILURE`